### PR TITLE
Design

### PR DESCRIPTION
**Pull request by MeganPegueros**
Description
What was done?
Implement design of the login and user registration parts. Likewise, javascript functionalities for a more interactive use.
Checkout more about the issue #9 

Why are these changes necessary?
The changes are necessary because they are the most important parts that constitute the web forum structurally.

What changes have been tested?
Changes to sass files were tested, as well as php files.

**List of the Changes**

Creation of structure and design: Creation of .php files by Ismael Montiel, including the logic and structure of the page, as well as the connection to the database.

### CODE REVIEW

File: resources/stylesheets/abstracts/_mixins.scss [hunk 1]
```diff
@@ -20,3 +20,33 @@ $breakpoint_tb: 800px;
...
+  display: grid;
+  align-items: center;
+  justify-content: center;
+}
+
+@mixin flex-left() {
+  display: flex;
+  justify-content: left;
+  align-items: center;
+}
```
Comment (Zama15) [inline resources/stylesheets/abstracts/_mixins.scss:52]: new line at the end of the file

File: resources/stylesheets/base/_login.scss [hunk 2]
```diff
@@ -0,0 +1,108 @@
+body {
+    background-image: url(./../../resources/img/background_prin.svg);
```
Comment (Zama15) [inline resources/stylesheets/base/_login.scss:2]: add in _variables.scss the path for the images

File: resources/stylesheets/base/_login.scss [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+.right {
+    border-radius: 0 0.9rem 0.9rem 0;
+    @include flex-grid;
+}
+
+.img{
+    width: 98%;
+    height: 100%;
+    border-radius: 0.5rem;
+}
```
Comment (Zama15) [inline resources/stylesheets/base/_login.scss:38]:
```
.img {
    width: 98%;
    height: 100%;
    border-radius: 0.5rem;
}
```
space between the class and the {

File: resources/stylesheets/base/_login.scss [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    &:hover {
+        color: $font_w;
+    }
+}
+
+.option {
+    @include flex-center;
+}
+
+.box-small{
```
Comment (Zama15) [inline resources/stylesheets/base/_login.scss:56]: space between the class and the {

File: resources/stylesheets/base/_login.scss [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    color: $font_w2;
+    font-size: 1rem;
+    margin: 0.5rem 0;
+}
+
+.principal {
+    @include flex-grid;
+    padding: 1.5rem 3rem;
+    border-radius: 2rem;
+}
```
Comment (Zama15) [inline resources/stylesheets/base/_login.scss:108]: new line at the end of the file

File: resources/stylesheets/base/_register.scss [hunk 6]
```diff
@@ -0,0 +1,105 @@
+*{
+    padding: 0;
+    margin: 0;
+}
```
Comment (Zama15) [inline resources/stylesheets/base/_register.scss:4]: this are called a reset styles. because it reset all the elements, so instead of doing in this file make a _reset.scss or use a _base.scss file inside the base folder to set the reset styles

File: resources/stylesheets/components/_alert.scss [hunk 7]
```diff
@@ -0,0 +1,48 @@
+.alert{
```
Comment (Zama15) [inline resources/stylesheets/components/_alert.scss:1]: space between the class and the {

File: resources/stylesheets/components/_alert.scss [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    color:$font_w1;
+    border-radius: 1rem;
+    margin: 0 0 0.9rem 0;
+    padding: 0 0 0 0.5rem;
+    font-weight: 700;
+    font-size: 1.1em;
+    width: 16.5rem;
+}
+
+.alert-success{
```
Comment (Zama15) [inline resources/stylesheets/components/_alert.scss:15]: space between the class and the {

File: resources/stylesheets/components/_alert.scss [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    margin: 0 0 2rem 0;
+    padding: 0.9rem;
+    font-weight: 700;
+    font-size: 1.3em;
+    width: 24rem;
+    border: 0.5rem solid $green3;
+    background-color: $green1;
+}
+
+.alert2{
```
Comment (Zama15) [inline resources/stylesheets/components/_alert.scss:29]: too ambiguos the name, the main thing you should think while naming a class is if everybody that see the code will understand without you present to explain it, not worry about the length of the name

File: resources/stylesheets/components/_alert.scss [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    /* color: rgb(211, 12, 12); */
+    border-radius: 1rem;
+    margin: 0 0 1.5rem 0;
+    padding: 0 0 0 0.5rem;
+    font-weight: 700;
+    font-size: 1.1em;
+    width: 20.5rem;
+}
+
+.icon-alert{
```
Comment (Zama15) [inline resources/stylesheets/components/_alert.scss:45]: space between the class and the {

File: resources/stylesheets/components/_cards.scss [hunk 11]
```diff
@@ -16,7 +16,7 @@
 
   .profile {
     cursor: pointer;
-    transition: $tr;   
+    /* transition: $tr;  */
```
Comment (Zama15) [inline resources/stylesheets/components/_cards.scss:19]: don't leave useless comments

File: resources/stylesheets/components/_cards.scss [hunk 12]
```diff
@@ -72,7 +72,7 @@
   p {
     margin-right: px(3); 
     cursor: pointer;
-    transition: $tr;   
+    /* transition: $tr; */
```
Comment (Zama15) [inline resources/stylesheets/components/_cards.scss:75]: don't leave useless comments

File: resources/stylesheets/components/_input.scss [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+    ::placeholder{
+        font-size: 1rem;
+        color: $brow2;
+        font-weight: 600;
+        
+    }
+}
+
+.icon{
```
Comment (Zama15) [inline resources/stylesheets/components/_input.scss:28]: space between the class and the {

File: resources/stylesheets/components/_nav.scss [hunk 14]
```diff
@@ -33,7 +33,7 @@ nav {
         align-items: center;
         padding: 8px;
         cursor: pointer;
-        transition: $tr; 
+        /* transition: $tr;  */
```
Comment (Zama15) [inline resources/stylesheets/components/_nav.scss:36]: don't leave useless comment

File: resources/stylesheets/components/_nav.scss [hunk 15]
```diff
@@ -54,7 +54,7 @@ nav {
       align-items: center;
       padding: 8px;
       cursor: pointer;
-      transition: $tr; 
+      /* transition: $tr; */
```
Comment (Zama15) [inline resources/stylesheets/components/_nav.scss:57]: don't leave useless comment

File: resources/stylesheets/components/_nav.scss [hunk 16]
```diff
@@ -130,12 +130,12 @@ nav {
   padding: 10px;
   cursor: pointer;
   font-size: 16px;
-  transition: $tr;
+  /* transition: $tr; */
```
Comment (Zama15) [inline resources/stylesheets/components/_nav.scss:133]: don't leave useless comment

File: resources/stylesheets/components/_nav.scss [hunk 17]
```diff
@@ -130,12 +130,12 @@ nav {
...
 }
 
 .subcategories .subcategory:hover {
   background-color: $font_w;
-  color: $font_g;
+  /* color: $font_g; */
```
Comment (Zama15) [inline resources/stylesheets/components/_nav.scss:138]: don't leave useless comment

**Review by Zama15 — COMMENTED** [review]
1. minors errors:
   - use a space between the class and the {
   - leave a new line at the end of the file
   - do not leave useless comments, comments are just for documentation 
> this comments are just to get used to the good practices, it doesn't affect the code or the functionality of the purpose
2. major errors:
   - the main purpose of the 7.1 patter folder to sass is to be aware of the scalability of the project I recommend use [this page](https://sass-guidelin.es/#naming-conventions) to watch out the purpose of each folder, the author do not follow the architecture of the folder and it is priority this error

**Review by Zama15 — CHANGES_REQUESTED** [review]
please fix

File: resources/stylesheets/abstracts/_mixins.scss [hunk 18]
```diff
@@ -20,3 +20,37 @@ $breakpoint_tb: 800px;
...
+
+@mixin flex-left() {
+  display: flex;
+  justify-content: left;
+  align-items: center;
+<<<<<<< HEAD
+}
+=======
+}
+>>>>>>> 601ef77768eb6cabd6c681cc03ee64578b30d86f
```
Comment (Zama15) [inline resources/stylesheets/abstracts/_mixins.scss:56]: why is this here?

File: resources/stylesheets/abstracts/_variables.scss [hunk 19]
```diff
@@ -1,23 +1,61 @@
-//whites
+<<<<<<< HEAD
+//back_image
+$back_img: url(./../../resources/img/background_prin.svg);
+
+=======
+>>>>>>> 601ef77768eb6cabd6c681cc03ee64578b30d86f
+//whitesb(119, 9, 9)
```
Comment (Zama15) [inline resources/stylesheets/abstracts/_variables.scss:7]: same here

**Review by Zama15 — CHANGES_REQUESTED** [review]
seems were some errors while pushing the changes, please check after the push if was correctly pushed. And remember read the [7.1 pattern](https://sass-guidelin.es/#naming-conventions), the folders are not being used correctly, some has to be in the page folders.

File: resources/stylesheets/page/_login.scss [hunk 20]
```diff
@@ -0,0 +1,108 @@
+body {
+    background-image: $back_img;
+    background-size: cover;
+    background-color: $theme_bl;
+    font-family: monospace;
+    @include flex-center;
+    width: 100vw;
+    height: 100vh;
+}
```
Comment (Zama15) [inline resources/stylesheets/page/_login.scss:9]: this seems major change to the body element, the main.scss will be apply to all the project, if this dont cause any error thru all the app, then move this to the _base.scss since the is use the same body for all the project

File: resources/stylesheets/page/_register.scss [hunk 21]
```diff
@@ -0,0 +1,105 @@
+*{
+    padding: 0;
+    margin: 0;
+}
```
Comment (Zama15) [inline resources/stylesheets/page/_register.scss:4]: this style is apply to all the elements so it goes to the base folder

File: resources/stylesheets/page/_register.scss [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+    font-size: 1rem;
+    margin: 0.5rem 0;
+}
+
+.option2 {
+    @include flex-center;
+    margin: 2rem 0 1rem 0;
+}
+
+.signup{
```
Comment (Zama15) [inline resources/stylesheets/page/_register.scss:57]: space between class and {

File: resources/stylesheets/page/_register.scss [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+    background-color: $brow3;
+    color: $font_w;
+    
+    &:hover {
+        color: $font_w;
+        background-color: $brow4;
+    }
+}
+
+.btn-create{
```
Comment (Zama15) [inline resources/stylesheets/page/_register.scss:76]: space between class and {

File: resources/stylesheets/page/_register.scss [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+    @include flex-center;
+}
+
+#message{
```
Comment (Zama15) [inline resources/stylesheets/page/_register.scss:80]: space between class and {

File: resources/stylesheets/page/_register.scss [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+    position: absolute;
+    bottom: -20px;
+    color:$font_w;
+    font-size: 15px;
+    display: none;
+    font-weight: 800;
+}
+
+#nomessage{
```
Comment (Zama15) [inline resources/stylesheets/page/_register.scss:89]: space between class and {

File: resources/stylesheets/page/_register.scss [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+    position: absolute;
+    bottom: -20px;
+    color:$font_w;
+    font-size: 15px;
+    font-weight: 800;
+    display: none;
+}
+
+#emailmsg{
```
Comment (Zama15) [inline resources/stylesheets/page/_register.scss:98]: space between class and {

File: resources/stylesheets/page/_verify.scss [hunk 27]
```diff
@@ -0,0 +1,67 @@
...
+    margin: 0.1rem 0;
+}
+
+.resend {
+    text-decoration: none;
+    color: $green2;
+    font-weight: 600;
+    padding-left: 0.2rem;
+
+    &:hover{
```
Comment (Zama15) [inline resources/stylesheets/page/_verify.scss:51]: space between class and {

**Review by Zama15 — CHANGES_REQUESTED** [review]
Minors were detected, some styles go in other folder and some style blocks needs a space following the best practices

File: resources/stylesheets/abstracts/_mixins.scss (hunk 1 above)
Comment (MeganPegueros) [reply resources/stylesheets/abstracts/_mixins.scss]: done

File: resources/stylesheets/base/_login.scss (hunk 3 above)
Comment (MeganPegueros) [reply resources/stylesheets/base/_login.scss]: done

File: resources/stylesheets/base/_login.scss (hunk 5 above)
Comment (MeganPegueros) [reply resources/stylesheets/base/_login.scss]: done

File: resources/stylesheets/base/_login.scss (hunk 4 above)
Comment (MeganPegueros) [reply resources/stylesheets/base/_login.scss]: done

File: resources/stylesheets/components/_input.scss [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+    &:focus{
+        background-color: $green1;
+        border: 0.1rem solid $green2;
+        outline: none;
+    }
+
+    &::placeholder{
+        color: $placeholder_1;
+    }
+}
```
[the remark this answers is not in the document]
Comment (MeganPegueros) [reply resources/stylesheets/components/_input.scss]: done

File: resources/stylesheets/components/_nav.scss (hunk 15 above)
Comment (MeganPegueros) [reply resources/stylesheets/components/_nav.scss]: okay

File: resources/stylesheets/components/_nav.scss (hunk 16 above)
Comment (MeganPegueros) [reply resources/stylesheets/components/_nav.scss]: okay

File: resources/stylesheets/components/_nav.scss (hunk 17 above)
Comment (MeganPegueros) [reply resources/stylesheets/components/_nav.scss]: okay

File: resources/stylesheets/abstracts/_mixins.scss (hunk 18 above)
Comment (MeganPegueros) [reply resources/stylesheets/abstracts/_mixins.scss]: done

File: resources/stylesheets/abstracts/_variables.scss (hunk 19 above)
Comment (MeganPegueros) [reply resources/stylesheets/abstracts/_variables.scss]: done

File: resources/stylesheets/page/_login.scss (hunk 20 above)
Comment (MeganPegueros) [reply resources/stylesheets/page/_login.scss]: Done, it shouldn´t cause any error.

File: resources/stylesheets/page/_register.scss (hunk 21 above)
Comment (MeganPegueros) [reply resources/stylesheets/page/_register.scss]: Done

**Review by AngelDiaz591 — APPROVED** [review]
